casper/src/test/scala/coop/rchain/casper/helper/HashSetCasperTestNode.scala

```diff
@@ -113,6 +111,19 @@ class HashSetCasperTestNode[F[_]](
           .void
       }
 
+  def addBlock(deployDatums: DeployData*): F[BlockMessage] =
+    for {
+      block <- createBlock(deployDatums: _*)
+      _     <- casperEff.addBlock(block, ignoreDoppelgangerCheck[F])
```

Comment:
A bit late but we should check that all addBlocks' return Valid unless they aren't. In other words,

```scala
status <- casperEff.addBlock(block, ignoreDoppelgangerCheck[F])
_ = status should be(Valid)
```